monai/transforms/post/array.py

```diff
@@ -289,6 +292,144 @@ def __call__(self, img: torch.Tensor) -> torch.Tensor:
+
+        Args:
+            img (NdarrayTensor): Pytorch tensor or numpy array of any shape.
+
+        Raises:
+            NotImplementedError: The provided image was not a Pytorch Tensor or numpy array.
+
+        Returns:
+            Union[np.ndarray, torch.Tensor]: Pytorch tensor or numpy array of the same shape as the input.
+        """
+        if isinstance(img, np.ndarray):
+            return np.asarray(np.where(np.isin(img, self.applied_labels), img, 0))
+        elif isinstance(img, torch.Tensor):
+            img_arr = img.detach().cpu().numpy()
+            img_arr = self(img_arr)
+            return torch.as_tensor(img_arr, device=img.device)
+        else:
+            raise NotImplementedError(f"{self.__class__} can not handle data of type {type(img)}.")
+
+
+class FillHoles(Transform):
+    r"""
+    This transform fills holes in the image and can be used to remove artifacts inside segments.
+
+    An enclosed hole is defined as a background pixel/voxel which is only enclosed by a single class.
+    The definition of enclosed can be defined with the connectivity parameter::
+
+        1-connectivity     2-connectivity     diagonal connection close-up
+
+             [ ]           [ ]  [ ]  [ ]             [ ]
+              |               \  |  /                 |  <- hop 2
+        [ ]--[x]--[ ]      [ ]--[x]--[ ]        [x]--[ ]
+              |               /  |  \             hop 1
+             [ ]           [ ]  [ ]  [ ]
+
+    It is possible to define for which labels the hole filling should be applied.
+    The input image is assumed to be a PyTorch Tensor or numpy array
+    with shape [batch_size, 1, spatial_dim1[, spatial_dim2, ...]] and the values correspond to expected labels.
+
+    Note:
+
+        The label 0 will be treated as background and the enclosed holes will be set to the neighboring class label.
+
+        The performance of this method heavily depends on the number of labels.
+        It is a bit faster if the list of `applied_labels` is provided.
+        Limiting the number of `applied_labels` results in a big decrease in processing time.
+
+    For example:
+
+        Use FillHoles with default parameters::
+
+            [1, 1, 1, 2, 2, 2, 3, 3]         [1, 1, 1, 2, 2, 2, 3, 3]
+            [1, 0, 1, 2, 0, 0, 3, 0]    =>   [1, 1 ,1, 2, 0, 0, 3, 0]
+            [1, 1, 1, 2, 2, 2, 3, 3]         [1, 1, 1, 2, 2, 2, 3, 3]
+
+        The hole in label 1 is fully enclosed and therefore filled with label 1.
+        The background label near label 2 and 3 is not fully enclosed and therefore not filled.
+    """
+
+    def __init__(
+        self, applied_labels: Optional[Union[Iterable[int], int]] = None, connectivity: Optional[int] = None
+    ) -> None:
+        """
+        Initialize the connectivity and limit the labels for which holes are filled.
+
+        Args:
+            applied_labels (Optional[Union[Iterable[int], int]], optional): Labels for which to fill holes. Defaults to None,
```

Comment:
As you already put type-hints in the code, no need to add the type in doc-strings.
Thanks.

Reply:
I can remove that but it is actually part of the doc standard to do that.